File: teste/foo.py
import math

count = 0

def foo(n):
    global count
    count += 1
    if n <= 1:
        return 1
    return foo(n / 4) + foo(n / 4)  # Aqui, n / 4 sempre será float

n = 10000
print(math.floor(2 * math.sqrt(n) - 1))
foo(n)
print(count)

File: teste/foo.cpp
#include <iostream>
#include <cmath>

using namespace std;

int count = 0;

int foo(int n) {
    count++;
    if (n <= 1) {
        return 1;
    }
    return foo(n / 4) + foo(n / 4);
}

int main() {
    int n = 10000;  // Agora n é inteiro
    cout << (2 * static_cast<int>(sqrt(n)) - 1) << endl;
    foo(n);
    cout << count << endl;
    return 0;
}
